Example/Stripe iOS Example (Custom)/SofortExampleViewController.m

```diff
@@ -79,31 +95,30 @@ - (void)pay {
         if (error) {
             [self.delegate exampleViewController:self didFinishWithError:error];
         } else {
-            self.source = source;
-            [self presentPollingUI];
-            [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(handleAppForeground) name:UIApplicationWillEnterForegroundNotification object:nil];
-            [[UIApplication sharedApplication] openURL:source.redirect.url];
-        }
-    }];
-}
-
-- (void)handleAppForeground {
-    [[NSNotificationCenter defaultCenter] removeObserver:self name:UIApplicationWillEnterForegroundNotification object:nil];
-    [[STPAPIClient sharedClient] startPollingSourceWithId:self.source.stripeID
-                                             clientSecret:self.source.clientSecret
-                                                  timeout:10
-                                               completion:^(STPSource *source, NSError *error) {
-        [self dismissPollingUI];
-        if (error) {
-            [self.delegate exampleViewController:self didFinishWithError:error];
-        } else {
-            if (source.status == STPSourceStatusConsumed) {
-                [self.delegate exampleViewController:self didFinishWithMessage:@"Payment successfully created"];
-            } else if (source.status == STPSourceStatusFailed) {
-                [self.delegate exampleViewController:self didFinishWithMessage:@"Payment failed"];
-            } else {
-                [self.delegate exampleViewController:self didFinishWithMessage:@"Order received"];
-            }
+            // In order to use STPRedirectContext, you'll need to set up
+            // your app delegate to forward URLs to the Stripe SDK.
+            // See `[Stripe handleStripeURLCallback:]`
+            STPRedirectContext *redirect = [[STPRedirectContext alloc] initWithSource:source completion:^(NSString *sourceID, NSString *clientSecret, NSError *error) {
+                [[STPAPIClient sharedClient] startPollingSourceWithId:sourceID
+                                                         clientSecret:clientSecret
+                                                              timeout:10
+                                                           completion:^(STPSource *source, NSError *error) {
+                                                               [self updateUIForPaymentInProgress:NO];
+                                                               if (error) {
+                                                                   [self.delegate exampleViewController:self didFinishWithError:error];
+                                                               } else {
+                                                                   if (source.status == STPSourceStatusConsumed ||
+                                                                       source.status == STPSourceStatusChargeable) {
```

Comment:
Should use switch statement instead of ifs so we get errors if cases are missing.